### ISSUE TITLE
project.assets.json' doesn't have a target for '.NETFramework,Version=v4.5.2'.

### ISSUE THREAD

**Issue by Saritha535** [googleads/googleads-dotnet-lib#146]
I added the AdwordsAPI V201802 project to my existing project then the below error came.  I am using  VS2017, How can I resolve this issue. Please help me out of this. 

project.assets.json' doesn't have a target for '.NETFramework,Version=v4.5.2'. Ensure that restore has run and that you have included 'net452' in the TargetFrameworks for your project.

**Comment by christopherseeley** [comment]
The project now uses nuget, so you'll need to use nuget in your project and run nuget restore. If you don't need to change any of the client library code, you can just add it as a nuget package instead of adding a project reference.

**Comment by Saritha535** [comment]
I already tried to restore the nuget package on my project even though its coming like that error.

**Comment by christopherseeley** [comment]
Does removing the bin and obj directories entirely and re-running restore help?

**Comment by Saritha535** [comment]
No. If I removed the bin and obj folders  again its restoring if I am building the solution.

**Comment by christopherseeley** [comment]
Can you clarify your exact setup?
You downloaded the AdWords project source from GitHub and are adding it as a project reference?

**Comment by Saritha535** [comment]
Yes, I downloaded the latest version of AdWords project from GitHub and I added Adwords.csproj file in to my project. Previously I used VS2015 the latest version of AdWords not supported. Then I installed the VS2017 and open the project at that time I facing this this. "project.assets.json' doesn't have a target for '.NETFramework,Version=v4.5.2'".
Please tell me the solution. I am facing this problem since one week and again March 28th is AdWords old versions will depreciate.

**Comment by AnashOommen** [comment]
I’m guessing that the error happens because the .net library targets both .net452 and .netstandard2.0, so Visual Studio will complain if you add reference directly to Adwords.csproj, but you are nit building against .net452.

Could you create a simple console project that’s identical to your project setup, and share the whole solution as an attachment? I wish to see how to get the csproj fixed for you.

Also, any reason why you don’t prefer using the nuget package?

**Comment by Saritha535** [comment]
I created sample project and added the reference of AdWords.csproj directly to my project.
Here I am attaching the sample console application. I am attaching the error that i got.
Please help me out this issue asap.

The reason behind that am not using nuget package is
From starting on wards in my project we are downloading the Ads Client library from git hub and adding .csproj file in my project. I am following those steps only. Once I tried to add from nuget package but I failed so after that I haven't tried of using nuget package to upgrading the AdWords APIs.

[Adwords_Analytics@saritha.zip](https://github.com/googleads/googleads-dotnet-lib/files/1814904/Adwords_Analytics.saritha.zip)

**Comment by Saritha535** [comment]
I am waiting for reply, please give any idea about this problem to solve the issue.
For AdWords API Depreciation on 28th March before that  I need to solve this issue.
Please help me out this problem.

**Comment by christopherseeley** [comment]
I downloaded your sample project, cloned the github source to the same relative path and the project built with no issues. Do you have the .NET Core SDK installed? This is now required for building the project from source.

Alternatively, you can use the binary distribution from the downloads section, or use the Google.AdWords nuget package.

**Comment by Saritha535** [comment]
Thank You..
Now I am not getting that error. Its build succeeded but I am getting run-time error. I have web reference in my project there I am getting below exception. Please cloud you assist me to resolve this issue. I attached my app.config file also.

SoapExtension System.Configuration.ConfigurationErrorsException in app.config The value of the property 'type' cannot be parsed

The below is the runtime exception:

**System.Configuration.ConfigurationErrorsException
  HResult=0x80131902
  Message=The value of the property 'type' cannot be parsed. The error is: Could not load type 'Google.Api.Ads.Common.Lib.SoapListenerExtension' from assembly 'Google.Ads.Common'.
  Source=<Cannot evaluate the exception source>
  StackTrace:<Cannot evaluate the exception stack trace>**


This is my web reference app.config code. I think I've tried all possible combinations for this line.
<system.web>
		<webServices>
			<soapExtensionTypes>
				<add type="Google.Api.Ads.Common.Lib.SoapListenerExtension, 
                                  Google.Ads.Common" priority="1" group="High" />
			</soapExtensionTypes>
		</webServices>
</system.web>
[App.config.zip](https://github.com/googleads/googleads-dotnet-lib/files/1828199/App.config.zip)

**Comment by jimper** [comment]
The SoapListenerExtension was removed in v22.0.0. Please see [this wiki article](https://github.com/googleads/googleads-dotnet-lib/wiki/Upgrading-your-client-library-to-v22.0.0) for details on how to update your App.config.

**Comment by Saritha535** [comment]
Thank you Jimper,
How to add dependency on System.ServiceModel.?

after commenting webservices related code I am getting the below error.

**Could not load file or assembly 'System.ServiceModel.Primitives, Version=4.2.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a' or one of its dependencies. The system cannot find the file specified.**
